keras_cv/metrics/coco/mean_average_precision.py

```diff
@@ -154,15 +159,19 @@ def update_state(self, y_true, y_pred, sample_weight=None):
         if isinstance(y_pred, tf.RaggedTensor):
             y_pred = y_pred.to_tensor(default_value=-1)
 
-        y_true = tf.cast(y_true, self.compute_dtype)
-        y_pred = tf.cast(y_pred, self.compute_dtype)
+        y_true = bounding_box.transform_format(
```

Comment:
Is the casting still needed here?

Reply:
Updated to do it internally, this is required for many conversions.